src/zenml/integrations/openai/hooks/open_ai_failure_hook.py

```diff
@@ -39,7 +40,9 @@ def openai_alerter_failure_hook_helper(
         exception: The exception that was raised.
         model_name: The OpenAI model to use for the chatbot.
     """
-    if context.stack and context.stack.alerter:
+    openai_api_key = os.environ.get("OPENAI_API_KEY")
+
+    if context.stack and context.stack.alerter and openai_api_key:
```

Comment:
do we need to check the alerter?

Reply:
If we don't check it and you don't have an alerter, later on when we run `context.stack.alerter.post(message)` this will raise an error.